### DISCUSSION TITLE
Social cards broke after opting in for the insiders program

### DISCUSSION THREAD

**Discussion by meysam81** [squidfunk/mkdocs-material#6979] [Q&A] [answered]
### Bug description

After joining the insiders program, my CI is now broken and I can't seem to figure out how to fix it. Does anyone have any clue what's missing?

https://github.com/developer-friendly/blog/actions/runs/8449090378/job/23142465169#step:9:89

### Related links

- https://squidfunk.github.io/mkdocs-material/insiders/

### Reproduction

Disabling social cards fixed the issue but I don't know what is still causing errors and have no clue on reproduction really

### Steps to reproduce

N/A

### Browser

Firefox

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by squidfunk** [comment]
Thanks for reporting! Since there's no reproduction, I'm converting this to a discussion, so somebody can help you debug your CI environment. If in the end there's indeed something wrong on our side, we're happy to fix it ☺️

**Reply by squidfunk** [reply]
Also note that we _need_ a reproduction, as we're not able to debug your CI build – we cannot make changes to your CI workflow, and without reproducing the problem on our machine, we can't fix it 😅

**Comment by meysam81** [comment, marked as answer]
For anyone else hitting the same issue, I managed to debug and solve the issue (after a couple of hours banging my head to the wall).




For some reason, not specifying the [`cards_layout_options.background_color`](https://squidfunk.github.io/mkdocs-material/plugins/social/#option.background_color) and  [`cards_layout_options.color`](https://squidfunk.github.io/mkdocs-material/plugins/social/#option.color) is a deal breaker for the social cards plugin, i.e., it does not revert to `theme.palette.primary` as documented in [here](https://squidfunk.github.io/mkdocs-material/plugins/social/#layouts).

Now, adding the background-color and color made it work:

https://github.com/developer-friendly/blog/commit/d0a7b22cc950591813b0f22ab8aec4aac92249a9

However, I saw some [ugly error message](https://github.com/developer-friendly/blog/actions/runs/8498617952/job/23278529331#step:8:24) I suspect is coming from the `mkdocs` itself.

cc @squidfunk

**Reply by squidfunk** [reply]
As already mentioned above: if you provide a minimal reproduction, we can fix it ☺️

**Reply by meysam81** [reply]
[9.5.16+insiders.4.53.4-social-cards-darkmode-conflict.zip](https://github.com/squidfunk/mkdocs-material/files/14818932/9.5.16%2Binsiders.4.53.4-social-cards-darkmode-conflict.zip)

**Reply by squidfunk** [reply]
Thanks for the reproduction! I can confirm that this crashes the social plugin. It stems from the fact that there is no [`primary`](https://squidfunk.github.io/mkdocs-material/setup/changing-the-colors/#primary-color) color included in the defined color palettes:

``` yaml
palette:
    - media: (prefers-color-scheme)
      toggle:
        icon: material/brightness-auto
        name: Switch to light mode
    - media: "(prefers-color-scheme: light)"
      scheme: default
      toggle:
        icon: material/brightness-7
        name: Switch to dark mode
    - media: "(prefers-color-scheme: dark)"
      scheme: slate
      toggle:
        icon: material/brightness-4
        name: Switch to system preference
```

Adding `primary` to the first palette fixes the issue. The problem is that the Jinja code filters the list of palettes down to an empty list, which the crashes the `first` filter. However, we can also relax the constraint of the social plugin a little, which will then just fall back to `indigo`, which we just did in [`b248bcf9c`](https://github.com/squidfunk/mkdocs-material-insiders/commit/b248bcf9c332018019bc3efdc30befc895ca17d3). The reproduction now runs without crashing.

Thanks again for taking the time to creating the reproduction and reporting this. You've helped us making Material for MkDocs more resilient, and thus easier to use, especially for non-technical folks ☺️
